pkg/sidekick/sidekick.go

```diff
@@ -0,0 +1,31 @@
+package sidekick
+
+import (
+	"context"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/dskit/services"
+
+	"github.com/grafana/pyroscope/pkg/frontend"
+	"github.com/grafana/pyroscope/pkg/objstore"
+	"github.com/grafana/pyroscope/pkg/sidekick/adhocprofiles"
+)
+
+type Sidekick struct {
+	services.Service
+
+	AdHocProfiles *adhocprofiles.AdHocProfiles
```

Comment:
I could make AdHocProfiles a subservice perhaps, if we decide that having a Sidekick module still makes sense. This could come in in a second PR where we "move" tenant settings here.